bin/describe.ml

```diff
@@ -593,23 +593,59 @@ module External_lib_deps = struct
     let to_dyn : t -> Dyn.t = function
       | Required -> String "required"
       | Optional -> String "optional"
-
-    let merge x y =
-      match (x, y) with
-      | Optional, Optional -> Optional
-      | _ -> Required
   end
 
   type external_lib_dep =
     { name : Lib_name.t
     ; kind : Kind.t
     }
 
-  type lib_deps =
-    { dir : Path.Source.t
-    ; deps : Lib_dep.t list
-    ; pps : Preprocess.With_instrumentation.t Preprocess.Per_module.t
-    }
+  let external_lib_dep_to_dyn t =
+    let open Dyn in
+    List [ String (Lib_name.to_string t.name); Kind.to_dyn t.kind ]
+
+  module Item = struct
+    module Kind = struct
+      type t =
+        | Executables
+        | Library
+        | Tests
+
+      let to_string = function
+        | Executables -> "executables"
+        | Library -> "library"
+        | Tests -> "tests"
+    end
+
+    type t =
+      { kind : Kind.t
+      ; dir : Path.Source.t
+      ; external_deps : external_lib_dep list
+      ; names : string list
+      ; package : Package.t option
+      }
+
+    let to_dyn t =
+      let open Dyn in
+      let names =
+        if t.kind = Kind.Library then ("name", String (List.hd t.names))
+        else ("names", (list string) t.names)
+      in
+      let package =
+        match t.package with
+        | Some p ->
```

Comment:
There are various functions that you can use like `Dyn.option`, `Package.Name.to_dyn`, etc.